system-contract-dapp-playground/src/components/contract-interaction/hts/shared/methods/handleAPIErrors.tsx

```diff
@@ -50,21 +51,25 @@ export const handleAPIErrors = ({
 }) => {
   const errorMessage = JSON.stringify(err);
 
-  let errorDescription = "See client's console for more information";
-  // @notice 4001 error code is returned when a metamask wallet request is rejected by the user
-  // @notice See https://docs.metamask.io/wallet/reference/provider-api/#errors for more information on the error returned by Metamask.
-  if (errorMessage.indexOf('4001') !== -1) {
-    errorDescription = 'You have rejected the request.';
-  } else if (errorMessage.indexOf('nonce has already been used') !== -1) {
-    errorDescription = 'Nonce has already been used. Please try again!';
-  } else if (errorMessage.indexOf('decreased allowance below zero') !== -1) {
-    errorDescription =
-      'The transaction was reverted due to the allowance decrease falling below zero.';
-  } else if (errorMessage.indexOf('transfer amount exceeds balance') !== -1) {
-    errorDescription = 'Transfer amount exceeds balance';
-  } else if (errorMessage.indexOf('insufficient allowance') !== -1) {
-    errorDescription = 'Insufficient allowance';
-  }
+  const reasonKeys = [
+    'NONCE',
+    'REJECT',
+    'INVALID_TOKENID',
+    'UNAUTHORIZED_CALLER',
+    'ALLOWANCE_BELOW_ZERO',
+    'INSUFFICIENT_ALLOWANCE',
+    'APPROVAL_CURRENT_CALLER',
+    'TRANSFER_EXCEEDS_BALANCE',
+  ];
+
+  let errorDescription = HEDERA_COMMON_WALLET_REVERT_REASONS.DEFAULT.description;
```

Comment:
Open a ticket to improve the handleAPIError() to map codes to error names.
This works for now but we can simplify later without special cases